### PR TITLE
Making the process of updating metadata a little bit more flexible

### PR DESCRIPTION
**Pull request by renatav**
**Fixes issue #**: Not applicable

**Description of the changes being introduced by the pull request**:

client/updater.py was slightly refactored in order to make the process of updating metadata files a bit more flexible. This flexibility is only related to how the files are downloaded and does not impact validation. Finding mirrors and the actual download were moved to a so-called handler class.
With that change, it is possible to implement the majority of the updater which would use a git repository to store metadata files. 

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: tuf/client/handlers.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      validation checks pass. In this case, nothing needs to be done,
+      so the method is empty.
+
+
+    <Arguments>
+      filename:
+        The relative file path (on the remote repository) of a metadata role.
+
+      mirror:
+        The mirror from whih th file was successfully downloaded.
```
Comment (danixeee) [inline tuf/client/handlers.py]: Typo

File: tuf/client/handlers.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    Side Effects>
+      None.
+
+    <Returns>
+      None.
+    """
+
+
+
+  def on_unsuccessful_update(self, filename):
```
Comment (danixeee) [inline tuf/client/handlers.py]: Does it make sense to move "hook" methods to `MetadataUpdater` class?

File: tuf/client/handlers.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    <Exceptions>
+      None.
+
+    Side Effects>
+      None.
+
+    <Returns>
+      None.
+    """
```
Comment (danixeee) [inline tuf/client/handlers.py]: Maybe we can log message that hook is called, also in `on_unsuccessful_update` method.

File: tuf/client/handlers.py (hunk 2 above)
Comment (renatav) [reply tuf/client/handlers.py]: Yeah, that would make sense. I think that I left them in the subclass in order to make my idea clearer

File: tuf/client/handlers.py (hunk 3 above)
Comment (renatav) [reply tuf/client/handlers.py]: I don't think that is necessary here. An error is logged in the `updater` class just after the `on_unsuccessful_update` is called. I didn't want to move it as well, as I think that that log call should be present in all implementation. So, we should not be able to remove it. We can log something more meaningful once we redefine the method.

**Review by danixeee — APPROVED** [review]
LGTM; merge when ready

**Pull request merged**